user-interface/src/main/java/life/qbic/datamanager/views/general/contact/AutocompleteContactField.java

```diff
@@ -0,0 +1,178 @@
+package life.qbic.datamanager.views.general.contact;
+
+import static java.util.Objects.isNull;
+
+import com.vaadin.flow.component.combobox.ComboBox;
+import com.vaadin.flow.component.customfield.CustomField;
+import com.vaadin.flow.component.html.Div;
+import com.vaadin.flow.component.html.Span;
+import com.vaadin.flow.data.binder.Binder;
+import com.vaadin.flow.data.renderer.ComponentRenderer;
+import com.vaadin.flow.data.validator.EmailValidator;
+import java.util.ArrayList;
+import java.util.List;
+import life.qbic.datamanager.views.general.HasBinderValidation;
+
+/**
+ * <b>A component for contact person input</b>
+ *
+ * <p>Provides components to add a contact person with a name and email.
+ * Includes methods for basic input validation.</p>
+ *
+ * @since 1.0.0
+ */
+public class AutocompleteContactField extends CustomField<Contact> implements
+    HasBinderValidation<Contact> {
+
+
+  private final ComboBox<Contact> contactSelection;
+
+  private final ComboBox<String> nameField;
+  private final ComboBox<String> emailField;
+  private final Binder<Contact> binder;
+
+  public AutocompleteContactField(String label) {
+    setLabel(label);
+    addClassName("contact-field");
+    binder = new Binder<>();
+    binder.addStatusChangeListener(event -> updateValidationProperty());
+
+    contactSelection = new ComboBox<>();
+    contactSelection.addClassName("contact-selection");
+    contactSelection.setAllowCustomValue(false);
+    contactSelection.setClearButtonVisible(true);
+    contactSelection.setRenderer(new ComponentRenderer<>(AutocompleteContactField::renderContact));
+    contactSelection.setItemLabelGenerator(
+        contact -> "%s - %s".formatted(contact.getFullName(), contact.getEmail()));
+
+    nameField = new ComboBox<>();
+    nameField.setAllowCustomValue(true);
+    nameField.addCustomValueSetListener(
+        customValueSet -> customValueSet.getSource().setValue(customValueSet.getDetail()));
+    nameField.setRequired(false);
+    nameField.addClassName("name-field");
+    nameField.setPlaceholder("Please enter a name");
+
+    emailField = new ComboBox<>();
+    emailField.setAllowCustomValue(true);
+    emailField.addCustomValueSetListener(
+        customValueSet -> customValueSet.getSource().setValue(customValueSet.getDetail()));
+    emailField.setRequired(false);
```

Comment:
Why is the email and nameField set as not Required here? 
I don't understand why we should allow to add a name without an email and vice versa 🤔

Reply:
this is done by the field itself :) not always required. the binder makes sure this does not happen